default/scripting/species/common/happiness.macros

```diff
@@ -69,84 +69,62 @@ COMMON_HAPPINESS_EFFECTS
             ]
 '''
 
+SUPPLY_CONNECTED_TO_CAPITAL_OR_REGAD
+'''
+		// this should ensure that there is a capital / admin for JUMPS_TO_CAPITAL_OR_REGAD
+        ResourceSupplyConnected empire = Source.Owner condition = And [
+            Or [Capital Building name = "BLD_REGIONAL_ADMIN"]
+            OwnedBy empire = Source.Owner
+        ]
+'''
+
+JUMPS_TO_CAPITAL_OR_REGAD
+'''
+		Statistic Min
+		    // Replace JumpsBetween by JumpsBetweenByEmpireSupplyConnections once it is implemented
+            value = JumpsBetween
+            object = Source.ID
+            object = LocalCandidate.ID
+            condition = And [
+                Or [Capital Building name = "BLD_REGIONAL_ADMIN"]
+                OwnedBy empire = Source.Owner
```

Comment:
why not use `SUPPLY_CONNECTED_TO_CAPITAL_OR_REGAD` here?

Reply:
Which part do you want to replace, the lines above your comment or the one line below?
The condition must identify the Capital and all RAs, but only if they are connected to the Source, i.e. the planet to be affected by the happiness modification. The Capital itself is of course always supply connected to the Capital.

Reply:
Actually, neither. I was misreading the structure of the condition in the macro above due to linebreaks, in part due to the tab characters causing a lot of indentation.

Reply:
Oh yeah, fun with tabs. At work we use an SVN hook to stop any check-in containing a tab.
The default settings for text editors in eclipse apparently create them. I changed the settings and updated the commit.